### PR TITLE
fix(driver): add support for alias.all when using dynamic alias

### LINKED ISSUE #25448: @alias.all Throws An Error When Defining Alias in Intercept With Route Handler in Version 12
### Current behavior

Currently, if you use `cy.intercept` with a route handler to define the alias, grabbing all instances of the alias `@alias.all` throws an error. This worked before upgrading to version 12 (verified it works with 11.2.0).

There error is:

```
cy.get() could not find a registered alias for: @alias.
You have not aliased anything yet.
```

### Desired behavior

Regardless of if you define an alias using `.as` or `request.alias`, you should still be able to access all of the alias instances using `@alias.all`.

### Test code to reproduce

I grabbed https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/e2e/commands/net_stubbing.cy.ts#L3678-L3691 and changed the way we alias the request to expose the issue.

```js
let routeCount = 0
const uniqueRoute = (route) => {
  routeCount += 1

  return `${route}-${routeCount}`
}

it('gets all aliased Interceptions by alias.all', function () {
    const url = uniqueRoute('/foo')

    cy.intercept(`${url}*`, (request) => {
      request.alias = 'alias';
      request.reply({ bar: 'baz' });
    })
      .then(() => {
        Cypress.$.get(url)
        Cypress.$.get(url)
      })
      .wait('@alias').wait('@alias')

      cy.get('@alias.all').then((interceptions) => {
        expect(interceptions).to.have.length(2)
      })
  })
  ```

### Cypress Version

12.*

### Node version

v16.19.0

### Operating System

Debian GNU/Linux 11 (bullseye)

### Debug Logs

_No response_

### Other

I was following https://github.com/cypress-io/cypress/issues/24653 and https://github.com/cypress-io/cypress/issues/25210 which seem to be related but not exactly the same. Since one issue was closed, I decided to raise this issue which shows a specific example that was working before upgrading to version 12.

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes https://github.com/cypress-io/cypress/issues/25448

### Additional details
Their is a bug in our driver where you can't `get` an alias using `.all` with dynamically assigned route aliases. 
```
cy.intercept(`/foo*`, (req) => {
    req.alias = 'alias'
})
```

You will get an error when attempting to get this alias with a `.all` like:
`cy.get('@dyanamicRouteAlias.all')`

This PR resolves this issue.

### Steps to test
Run the following test in the `packages/driver/cypress/e2e/commands/net_stubbing.cy.ts` => `gets all aliased Interceptions by alias.all when assigning an alias using req.alias`

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,14 @@ _Released 11/21/2023 (PENDING)_
 
 - When artifacts are uploaded to the Cypress Cloud, the duration of each upload will be displayed in the console. Addresses [#28237](https://github.com/cypress-io/cypress/issues/28237).
 
+**Bugfixes:**
+
+- Fixed an issue getting a dynamic route alias using .all. Addresses [#25448](https://github.com/cypress-io/cypress/issues/25448)
```
Comment (mschile) [inline cli/CHANGELOG.md]:
```suggestion
- Fixed an issue getting a dynamic route alias using `@alias.all`. Addresses [#25448](https://github.com/cypress-io/cypress/issues/25448)
```

**Review by mschile — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -7,10 +7,14 @@ _Released 11/21/2023 (PENDING)_
...
 
+**Bugfixes:**
+
+- Fixed an issue getting a dynamic route alias using `@alias.all`. Addresses [#25448](https://github.com/cypress-io/cypress/issues/25448)
+
 **Misc:**
 
 - Browser tabs and windows other than the Cypress tab are now closed between tests in Chromium-based browsers. Addressed in [#28204](https://github.com/cypress-io/cypress/pull/28204).
-- Cypress now ensures the main browser tab is active before running eaech command in Chromium-based browsers. Addressed in [#28334](https://github.com/cypress-io/cypress/pull/28334).
+- Cypress now ensures the main browser tab is active before running each command in Chromium-based browsers. Addressed in [#28334](https://github.com/cypress-io/cypress/pull/28334).
```
Comment (chrisbreiding) [inline cli/CHANGELOG.md:18]: Oops 😅 Thanks!

File: packages/driver/cypress/e2e/commands/net_stubbing.cy.ts [hunk 3]
```diff
@@ -3699,6 +3699,25 @@ describe('network stubbing', { retries: 15 }, function () {
       .wait('@a').its('response.body').should('eq', 'bar')
     })
 
+    // @see https://github.com/cypress-io/cypress/issues/25448
+    it('gets all aliased Interceptions by alias.all when assigning an alias using req.alias', function () {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/e2e/commands/net_stubbing.cy.ts]: Can we move this into the ` context('cy.get(alias)', function () {` context next to the existing test `it('gets all aliased Interceptions by alias.all', function () {`?

File: cli/CHANGELOG.md (hunk 2 above)
Comment (jordanpowell88) [reply cli/CHANGELOG.md:18]: haha! You're welcome. Thought I'd fix it while I was in there. Been there before

**Review by chrisbreiding — APPROVED** [review]

File: packages/driver/cypress/e2e/commands/net_stubbing.cy.ts (hunk 3 above)
Comment (jordanpowell88) [reply packages/driver/cypress/e2e/commands/net_stubbing.cy.ts]: Yup! Done in 9682cbbdbc9bb6a40856a7861196f01d7e45b5f0

File: cli/CHANGELOG.md [hunk 4]
```diff
@@ -7,10 +7,14 @@ _Released 11/21/2023 (PENDING)_
 
 - When artifacts are uploaded to the Cypress Cloud, the duration of each upload will be displayed in the console. Addresses [#28237](https://github.com/cypress-io/cypress/issues/28237).
 
+**Bugfixes:**
+
+- Fixed an issue getting a dynamic route alias using `@alias.all`. Addresses [#25448](https://github.com/cypress-io/cypress/issues/25448)
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
- Fixed an issue where [aliasing individual requests](https://docs.cypress.io/api/commands/intercept#Aliasing-individual-requests) with `cy.intercept()` led to an error when retrieving all of the aliases with `cy.get(@alias.all)` . Addresses [#25448](https://github.com/cypress-io/cypress/issues/25448)
```

something a bit more insightful to the bug is needed

**Pull request merged**